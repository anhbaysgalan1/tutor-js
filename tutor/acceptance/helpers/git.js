const gitP = require('simple-git/promise');
const path = require('path');
const fs = require('fs-extra');
const { imagesDir } = require('./images-dir');
const chalk = require('chalk');

const repoURL = 'https://github.com/openstax/tutor-images.git';

function initialiseRepo(git) {
  console.log(chalk.yellow(`Cloning images repo to ${imagesDir}`));
  return git.clone(repoURL, imagesDir);
}

function getCurrentBranch() {
  const git = gitP(__dirname);
  return new Promise((resolve, reject) => {
    git.branchLocal((err, branch) => {
      if (err) { reject(err); }
      resolve(branch.current);
    });
  });
}

async function gitCommit() {
  const g = git();
  const currentBranchName = await getCurrentBranch();
  const branchInfo = await g.branch();
  if (branchInfo.all.includes(currentBranchName)) {
    await g.checkout(currentBranchName);
  } else {
    await g.checkoutLocalBranch(currentBranchName);
  }
  await g.add(['--all', '.']);
  await g.commit(`updated ${currentBranchName}`);
  await g.push('origin', currentBranchName);
}

async function gitInit() {
  await fs.ensureDir(imagesDir);
  const git = gitP(imagesDir);
  return git.cwd(imagesDir)
    .then(() => git.checkIsRepo())
    .then(isRepo => isRepo ? git : initialiseRepo(git))
    .then(() => git.fetch())
    .then(() => git);
}

function git() {
  process.exit(1)
  return gitP(imagesDir);
}

module.exports = { git, gitCommit, gitInit, getCurrentBranch };
