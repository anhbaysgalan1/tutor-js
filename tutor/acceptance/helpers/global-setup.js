const Server = require('./server');
const puppeteer = require('puppeteer');
const fs = require('fs-extra');
const del = require('del');
const path = require('path');
const { gitInit, gitCommit } = require('./git');
const repoURL = 'https://github.com/openstax/tutor-images.git';

const DIR = require('./working-directory');

async function boot () {
  await gitInit();
  const browser = await puppeteer.launch();
  global.__BROWSER__ = browser;
  fs.ensureDirSync(DIR);
  fs.writeFileSync(path.join(DIR, 'wsEndpoint'), browser.wsEndpoint());

  global.global.__SERVER__ = new Server();
  await global.__SERVER__.boot();
}

module.exports = boot;
