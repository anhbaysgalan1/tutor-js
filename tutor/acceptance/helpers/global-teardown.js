const { gitCommit } = require('./git');
const Server = require('./server');
const DIR = require('./working-directory');
const fs = require('fs-extra');

async function teardown () {
  await global.__BROWSER__.close();
  await global.__SERVER__.halt();
  fs.removeSync(DIR);
  return gitCommit();
}

module.exports = teardown;
