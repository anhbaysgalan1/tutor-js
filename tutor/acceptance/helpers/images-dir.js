const imagesDir = process.env.SCREENSHOTS_DIR;

module.exports = { imagesDir };
