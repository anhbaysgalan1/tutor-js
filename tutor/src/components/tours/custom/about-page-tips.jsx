import React from 'react';
import { action } from 'mobx';
import { observer, inject } from 'mobx-react';

import TourContext from '../../../models/tour/context';

import { Tooltip } from 'react-joyride';

@inject((allStores, props) => ({ tourContext: ( props.tourContext || allStores.tourContext ) }))
@observer
export default class AboutPageTips extends React.PureComponent {
  // static propTypes = {
  //   tourContext: React.PropTypes.instanceOf(TourContext)
  // }

  @action.bound
  handleClick(clickEvent) {
    let parentRide = this.props.ride.joyrideRef;

    const { index, shouldRun } = parentRide.state;
    const { steps, type } = parentRide.props;
    const el = clickEvent.currentTarget.className.includes('joyride-') && [
      'A',
      'BUTTON'
    ].includes(clickEvent.currentTarget.tagName) ? clickEvent.currentTarget : clickEvent.target;
    const dataType = el.dataset.type;

    /* istanbul ignore else */
    if (el.className.indexOf('joyride-') === 0 && dataType === 'back') { 
      clickEvent.preventDefault();
      clickEvent.stopPropagation();
      parentRide.stop();
      parentRide.reset();
      this.props.tourContext.playTriggeredTours();
    } else {
      parentRide.onClickTooltip(clickEvent);
    }
  }

  render () {
    return (
      <Tooltip
        buttons={{primary: 'View later', secondary: 'View tips now'}}
        onClick={this.handleClick}
        {...this.props}
      />
    );
  }
}
