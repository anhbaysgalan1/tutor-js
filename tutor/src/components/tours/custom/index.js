import HowToUseQL               from './how-to-use-ql';
import HowToUsePreview          from './how-to-use-preview';
import ViewedOnBlur             from './viewed-on-blur';
import CenteredWheel            from './centered-wheel';
import CenteredNoHoleWheel      from './centered-no-hole-wheel';
import HowToBuildYourReading    from './how-to-build-your-reading';
import ForceScrollOnMount       from './force-scroll-on-mount';
import TipsNowOrLater           from './tips-now-or-later';

import WelcomeToTutor               from './welcome-to-tutor';
import WelcomeToTutorWithCoach      from './welcome-to-tutor-with-coach';
import LookingForCoach              from './looking-for-coach';
import LookingForCoachNoMigration   from './looking-for-coach-no-migration';

export default {
  HowToUseQL,
  HowToUsePreview,
  ViewedOnBlur,
  CenteredWheel,
  CenteredNoHoleWheel,
  HowToBuildYourReading,
  ForceScrollOnMount,
  TipsNowOrLater,

  WelcomeToTutor,
  WelcomeToTutorWithCoach,
  LookingForCoach,
  LookingForCoachNoMigration,
};
