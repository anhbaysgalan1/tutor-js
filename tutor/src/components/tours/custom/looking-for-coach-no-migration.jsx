import React from 'react';

import {
  Column,
  TutorCoachSunset,
} from './common';
import CourseBranding from '../../branding/course';
import SuperTrainingWheel from './super-training-wheel';

export default class LookingForCoachNoMigration extends React.PureComponent {

  render () {
    return (
      <SuperTrainingWheel {...this.props} className='has-forest-background'>
        <TutorCoachSunset>
          <Column className="not-launched-yet">
            <p>We haven't launched<br/>
              OpenStax Tutor in your subject yet. We'll let you know as soon as it's available.</p>
          </Column>
        </TutorCoachSunset>
      </SuperTrainingWheel>
    );
  }
}
