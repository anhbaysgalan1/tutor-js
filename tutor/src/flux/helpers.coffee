_ = require 'underscore'
flux = require 'flux-react'

LOADING = 'loading'
LOADED  = 'loaded'
FAILED  = 'failed'
SAVING  = 'saving'
DELETING = 'deleting'
DELETED = 'deleted'

STATES =
  LOADING : LOADING
  LOADED  : LOADED
  FAILED  : FAILED
  SAVING  : SAVING
  DELETING: DELETING
  DELETED : DELETED

idCounter = 0
CREATE_KEY = -> "_CREATING_#{idCounter++}"

isNew = (id) -> /_CREATING_/.test(id)

CrudConfig = ->
  {
    _asyncStatus: {}
    _local: {}
    _changed: {}
    _errors: {}
    _reload: {}

    # If the specific type needs to do something else to the object:
    # _loaded : (obj, id) ->
    # _saved : (obj, id) ->
    # _reset : ->

    reset: ->
      @_asyncStatus = {}
      @_local = {}
      @_changed = {}
      @_errors = {}
      @_reload = {}

      @_reset?()
      @emitChange()

    dontReload: (id) ->
      @_asyncStatus[id] is LOADED and @_HACK_DO_NOT_RELOAD

    FAILED: (status, msg, id, args...) ->
      @_asyncStatus[id] = FAILED
      @_errors[id] = msg
      @_failed?({status, msg}, id, args...)

      unless status is 0 # indicates network failure
        delete @_local[id]
        @emitChange()

    load: (id) ->
      @_load(id)

    # allow aliasing load without triggering api call.
    _load: (id) ->
      # Add a shortcut for unit testing
      return if @dontReload(id)
      @_reload[id] = false
      @_asyncStatus[id] = LOADING
      @emitChange()

    loaded: (obj, id, args...) ->
      # id = obj.id
      @_asyncStatus[id] = LOADED

      if obj
        # If the specific type needs to do something else to the object:
        @_local[id] = @_loaded?(obj, id, args...) or obj

      @emitChange()

    save: (id, obj) ->
      # Note: id could be isNew()
      @_asyncStatus[id] = SAVING
      @emitChange()

    saved: (result, id, args...) ->
      # id = result.id
      @_asyncStatus[id] = LOADED # TODO: Maybe make this SAVED

      # If the specific type needs to do something else to the object:
      obj = @_saved?(result, id, args...)
      result = obj if obj

      if result
        @_local[id] = result
        @_local[result.id] = result
        delete @_changed[result.id]
      else
        console.warn('API WARN: Server did not return JSON after saving. Patching locally')
        # Merge all the local changes into the new local object
        @_local[id] = _.extend(@_local[id], @_changed[id])
      delete @_changed[id]
      delete @_errors[id]
      # If the specific type needs to do something else to the object:
      @emitChange()

    create: (localId, attributes = {}) ->
      throw new Error('BUG: MUST provide a local id') unless isNew(localId)
      @_local[localId] = {}
      @_changed[localId] = attributes
      @_asyncStatus[localId] = LOADED

    created: (result, localId, args...) ->
      @_local[localId] = result # HACK: So react component can still manipulate the same object
      @_local[result.id] = result
      @_asyncStatus[localId] = LOADED
      @_asyncStatus[result.id] = LOADED
      @emitChange()
      @_created?(result, localId, args...)

    _change: (id, obj) ->
      @_changed[id] ?= {}
      _.extend(@_changed[id], obj)
      @emitChange()

    _save: (id) ->
      @_asyncStatus[id] = SAVING

    delete: (id) ->
      @_asyncStatus[id] = DELETING
      @_delete?(id)

    deleted: (result, id, args...) ->
      @_asyncStatus[id] = DELETED
      delete @_local[id]
      @emitChange()
      @_deleted?(result, id, args...)

    clearChanged: (id) ->
      delete @_changed[id]

    HACK_DO_NOT_RELOAD: (bool) -> @_HACK_DO_NOT_RELOAD = bool

    # Keep this here so other exports method have access to it
    _get: (id) ->
      val = @_local[id]
      return null unless val or @_asyncStatus[id] is SAVING
      # Scores stores an Array unlike most other stores
      if _.isArray(val)
        val
      else
        _.extend({}, val, @_changed[id])

    exports:
      isUnknown: (id) -> not @_asyncStatus[id]
      isLoading: (id) -> @_asyncStatus[id] is LOADING
      isLoaded: (id) -> @_asyncStatus[id] is LOADED
      isDeleting: (id) -> @_asyncStatus[id] is DELETING
      isSaving: (id) -> @_asyncStatus[id] is SAVING
      isFailed: (id) -> @_asyncStatus[id] is FAILED
      getAsyncStatus: (id) -> @_asyncStatus[id]
      get: (id) -> @_get(id)
      isChanged: (id) -> not _.isEmpty(@_changed[id])
      getChanged: (id) -> @_changed[id] or {}
      freshLocalId: -> CREATE_KEY()
      isNew: (id) -> isNew(id)
      reload: (id) -> @_reload[id]
      getError: (id) -> @_errors[id]

  }

# Helper for creating a simple store for actions
makeSimpleStore = (storeConfig) ->

  actionsConfig = _.omit storeConfig, (value, key) ->
    typeof value isnt 'function' or key is 'exports'

  actionsConfig = _.keys(actionsConfig)
  actions = flux.createActions(actionsConfig)
  storeConfig.actions = _.values(actions)
  store = flux.createStore(storeConfig)
  {actions, store}


extendConfig = (newConfig, origConfig) ->
  newConfig.exports ?= {}
  _.defaults(newConfig, origConfig)
  _.defaults(newConfig.exports, origConfig.exports)
  newConfig


makeStandardStore = (name, storeConfig) ->
  extendConfig(storeConfig, new CrudConfig())
  {actions, store} = makeSimpleStore(storeConfig)
  {"#{name}Actions":actions, "#{name}Store":store}


module.exports = {CrudConfig, makeSimpleStore, extendConfig, makeStandardStore, isNew, STATES}
