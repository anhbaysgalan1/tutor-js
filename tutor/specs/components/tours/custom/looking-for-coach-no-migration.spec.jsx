import {SnapShot} from '../../helpers/component-testing'
import { LookingForCoachNoMigrationContent } from '../../../../src/components/tours/custom/looking-for-coach-no-migration';

describe('Looking for Coach no migration', () => {
  it('matches snapshot for looking for coach without migration', () => {
    expect(SnapShot.create(<LookingForCoachNoMigrationContent step={{ride:{}}} />).toJSON()).toMatchSnapshot();
  });

});
