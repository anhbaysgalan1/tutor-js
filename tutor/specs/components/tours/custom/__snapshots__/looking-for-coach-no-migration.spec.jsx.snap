// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Looking for Coach no migration matches snapshot for looking for coach without migration 1`] = `
<div
  className="value-prop cc-sunset"
>
  <h1
    className="heading"
  >
    Looking for your Concept Coach courses?
  </h1>
  <div
    className="column-content"
  >
    <div
      className="column thanks"
    >
      <p>
        Thanks for participating in the
        <br />
        Concept Coach pilot! Read our
        <br />
        <a
          href="https://openstax.org/blog/concept-coach-ending-greater-tools-are-coming"
          target="_blank"
        >
          blog post
        </a>
         to find out what we learned and how weâ€™re moving forward.
      </p>
    </div>
    <div
      className="column export-by"
    >
      <p>
        The last day to export your
        <br />
        Concept Coach scores reports is October 1.
      </p>
    </div>
    <div
      className="column not-launched-yet"
    >
      <p>
        We haven't launched
        <br />
        OpenStax Tutor in your subject yet. We'll let you know as soon as it's available.
      </p>
    </div>
  </div>
</div>
`;
